server/src/main/java/org/opensearch/search/fetch/subphase/MatchedQueriesPhase.java

```diff
@@ -71,35 +69,43 @@ public FetchSubPhaseProcessor getProcessor(FetchContext context) throws IOExcept
         for (Map.Entry<String, Query> entry : namedQueries.entrySet()) {
             weights.put(
                 entry.getKey(),
-                context.searcher().createWeight(context.searcher().rewrite(entry.getValue()), ScoreMode.COMPLETE_NO_SCORES, 1)
+                context.includeNamedQueriesScore()
+                    ? context.searcher().createWeight(context.searcher().rewrite(entry.getValue()), ScoreMode.COMPLETE, 1)
+                    : context.searcher().createWeight(context.searcher().rewrite(entry.getValue()), ScoreMode.COMPLETE_NO_SCORES, 1)
             );
         }
         return new FetchSubPhaseProcessor() {
```

Comment:
I think the @msfroh idea was to have 2 `getProcessor` method implementation, depending on `context.includeNamedQueriesScore() = true | false` , that would be cleaner and easier to maintain vs deciphering conditional logic

Reply:
👍 let me refactor it a bit

Reply:
i do believe we still have to define the score mode based on the includeNamedQueriesScore condition

Reply:
`ScoreMode scoreMode = context.includeNamedQueriesScore() ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;`